### PR TITLE
update retries mechanism

### PR DESCRIPTION
**Pull request by acidmk**

### CODE REVIEW

File: request.go [hunk 1]
```diff
@@ -227,6 +231,14 @@ func (c *Client) buildURL(path string, opts requestOpts) (string, error) {
 	return fullURL.String(), nil
 }
 
+func checkForRetry(resp *http.Response, err error) (bool, error) {
```
Comment (dshemin) [inline request.go:234]: Argument `err` didn't used here.

File: solus.go [hunk 2]
```diff
@@ -106,7 +107,13 @@ func WithLogger(logger Logger) ClientOption {
 }
 
 // NewClient create and initialize Client instance.
-func NewClient(baseURL *url.URL, a Authenticator, opts ...ClientOption) (*Client, error) {
+func NewClient(
+	baseURL *url.URL,
+	a Authenticator,
+	retries int,
```
Comment (dshemin) [inline solus.go:113]: Please use `ClientOption` instead of adding two new arguments.

**Review by dshemin — CHANGES_REQUESTED** [review]

File: request.go (hunk 1 above)
Comment (acidmk) [reply request.go:234]: now handling error

File: solus.go (hunk 2 above)
Comment (acidmk) [reply solus.go]: added `SetRetryPolicy` option to set both `Retry` and `RetryAfter`

**Review by dshemin — APPROVED** [review]

**Pull request merged**